neqo-http3/src/connection_client.rs

```diff
@@ -822,15 +822,20 @@ impl Http3Client {
     /// packets need to be sent or if a timer needs to be updated.
     ///
     /// [1]: ../neqo_transport/enum.ConnectionEvent.html
-    pub fn process_input(&mut self, dgram: Datagram, now: Instant) {
+    pub fn process_input(&mut self, dgram: &Datagram, now: Instant) {
         qtrace!([self], "Process input.");
         self.conn.process_input(dgram, now);
         self.process_http3(now);
     }
 
-    pub fn process_multiple_input(&mut self, dgrams: Vec<Datagram>, now: Instant) {
+    pub fn process_multiple_input<'a>(
+        &mut self,
+        // TODO: Need ExactSizeIterator for qtrace below. Worth the type restriction?
+        dgrams: impl ExactSizeIterator<Item = &'a Datagram>,
```

Comment:
Hmm, I might prefer instead `I: IntoIterator, I::IntoIter: ExactSizeIterator, I::Item: &'a Datagram` in this case.  Then the caller can pass a slice.

Reply:
Good idea. Thank you @martinthomson. See 3f76accc.